File: apps/macos/OvaFlusDesktop/Models/Budget.swift
import Foundation

struct Budget: Identifiable, Codable, Hashable {
    let id: String
    let name: String
    let category: String
    let limit: Double
    var spent: Double
    let period: BudgetPeriod
    let startDate: Date
    let endDate: Date
    let createdAt: Date
    let updatedAt: Date

    var remaining: Double { limit - spent }
    var percentUsed: Double { limit > 0 ? spent / limit * 100 : 0 }
    var isOverBudget: Bool { spent > limit }

    enum BudgetPeriod: String, Codable {
        case weekly
        case monthly
        case yearly
    }
}

File: packages/shared-types/src/budget.ts
export interface Budget {
  id: string;
  userId: string;
  name: string;
  description?: string;
  timeframe: 'weekly' | 'monthly' | 'custom';
  startDate: Date;
  endDate: Date;
  totalLimit: number;
  currency: string;
  categories: BudgetCategory[];
  createdAt: Date;
  updatedAt: Date;
}

export interface BudgetCategory {
  id: string;
  budgetId: string;
  name: string;
  color: string;
  icon: string;
  limit: number;
  spent: number;
  subcategories: BudgetSubcategory[];
}

export interface BudgetSubcategory {
  id: string;
  categoryId: string;
  name: string;
  limit: number;
  spent: number;
}

export interface BudgetAlert {
  id: string;
  budgetId: string;
  categoryId?: string;
  threshold: number; // 0-100
  triggered: boolean;
  triggeredAt?: Date;
}

export interface BudgetTemplate {
  id: string;
  name: string;
  description?: string;
  categories: Omit<BudgetCategory, 'id' | 'budgetId' | 'spent'>[];
}
